mage_ai/data_preparation/logging/logger_manager.py

```diff
@@ -0,0 +1,115 @@
+from dataclasses import dataclass, field
+from mage_ai.data_preparation.models.constants import LOGS_DIR
+from mage_ai.data_preparation.models.file import File
+from mage_ai.data_preparation.repo_manager import RepoConfig, get_repo_path
+from mage_ai.shared.array import find
+from mage_ai.shared.config import BaseConfig
+from typing import Dict
+import io
+import logging
+import os
+import sys
+    
+
+@dataclass
+class LoggingConfig(BaseConfig):
+    type: str
+    level: str = 'INFO'
+    destination_config: Dict = field(default_factory=dict)
+
+
+class LoggerManager:
+    def __init__(
+        self,
+        repo_path: str = None,
+        logs_dir: str = None,
+        pipeline_uuid: str = None,
+        block_uuid: str = None,
+        partition: str = None,
+        repo_config: RepoConfig = None,
+    ):
+        self.repo_path = repo_path
+        self.logs_dir = logs_dir
+        self.pipeline_uuid = pipeline_uuid
+        self.block_uuid = block_uuid
+        self.partition = partition
+
+        self.repo_config = repo_config
+        logging_config = self.repo_config.logging_config if self.repo_config else dict()
+        self.logging_config = LoggingConfig.load(config=logging_config)
+
+        logger_name_parts = [self.pipeline_uuid]
+        if self.partition is not None:
+            logger_name_parts.append(self.partition)
+        if self.block_uuid is not None:
+            logger_name_parts.append(self.block_uuid)
+        logger_name = '/'.join(logger_name_parts)
+
+        self.logger = logging.getLogger(logger_name)
+
+        self.log_level = logging.getLevelName(self.logging_config.level)
+        self.logger.setLevel(self.log_level)
+
+        self.formatter = logging.Formatter(
+            '%(asctime)s %(message)s',
+            '%Y-%m-%dT%H:%M:%S',
+        )
+        self.stream = None
+        if not self.logger.handlers:
+            if self.logging_config.destination_config:
```

Comment:
Should this only be applied to s3 logging?

Reply:
i was thinking if they specify any destination config, then we would need to pass the logs to the stream because we would need to output the logs to the specified destination

Reply:
ok